### PR TITLE
test: add the unit test and comments for ylm.h

### PR DESCRIPTION
**Pull request by pxlxingliang**
range: source/module_base

### CODE REVIEW

File: source/module_base/test/math_ylmreal_test.cpp [hunk 1]
```diff
@@ -86,7 +195,10 @@ class YlmRealTest : public testing::Test
         g[3].set(-1.0,-1.0,-1.0);
 
         rly = new double[nylm];
-        ref = new double[nylm*ng]{
+        rlyvector.resize(nylm);
+        rlgy = new double[nylm][3];
+        rlgyvector.resize(nylm,std::vector<double>(3));
+        ref = new double[64*4]{
```
Comment (caic99) [inline source/module_base/test/math_ylmreal_test.cpp:201]: It seems that all variables in `SetUp`  is of constant value.

Comment (pxlxingliang) [reply source/module_base/test/math_ylmreal_test.cpp:201]: yes

Comment (caic99) [reply source/module_base/test/math_ylmreal_test.cpp:201]: > yes

If the allocated arrays are not been used in the following part for pointer usage (e.g. realloc), it is better to put them in class.

Comment (caic99) [reply source/module_base/test/math_ylmreal_test.cpp:201]: https://stackoverflow.com/questions/27408819/why-cant-non-static-data-members-be-constexpr
@pxlxingliang FYI

**Review by caic99 — APPROVED** [review]

**Review by dyzheng — APPROVED** [review]

**Pull request merged**